lib/resty/core/request.lua

```diff
@@ -310,16 +314,53 @@ do
 
         local rc
         if not value then
-            rc = C.ngx_http_lua_ffi_req_header_set_single_value(r, name,
-                                                         #name, nil, 0)
+            if not override then
+                error("invalid header value", 3)
```

Comment:
I think we should be more descriptive here. We usually are in our other APIs... How about: `"bad value arg: string or table expected, got nil"`. We should also update the `if not value` condition above to better handle truthy/falsy values. So maybe change it to `if value == nil`? The condition in `ngx_resp.add_header()` does this already.